File: CheckforRotation.c
#include<stdio.h>
#include<string.h>
#include<stdbool.h>

bool areRotations( char s1[] , char s2[]){
    int n = strlen(s1);

    for( int i = 0 ; i < n ; ++i){
        if(strcmp( s1, s2) == 0 )
        return true ;

        char last = s1[n -1 ];
        for( int j = n - 1 ; j > 0 ; j--){
            s1[j] = s1[ j - 1];
        }
        s1[0] = last ;

    }
    return false ;
}

int main()
{
    char s1[] = "aab";
    char s2[] = "aba";

    printf("%s" , areRotations(s1 , s2) ? "true" : "false");
    return 0;
}

File: CheckforRotation.java
public class CheckforRotation {
    static boolean areRotations(String s1 , String s2){

        int n = s1.length();

        for( int i = 0 ; i < n ; ++i){
            if(s1.equals(s2)){
                return true ;
            }
            char last = s1.charAt(s1.length() - 1 );
            s1 = last + s1.substring( 0 , s1.length() - 1 );

        }
        return false ;

    }

    public static void main(String [] args ){
        String s1 = "aab";
        String s2 = "aab";

        System.out.println(areRotations( s1 , s2 ));
    }
}
